src/plugins/data_source_management/public/types.ts

```diff
@@ -76,20 +50,34 @@ export type DataSourceManagementContextValue = OpenSearchDashboardsReactContextV
   DataSourceManagementContext
 >;
 
-export enum CredentialSourceType {
-  CreateCredential = 'createCredential',
-  ExistingCredential = 'existingCredential',
-  NoAuth = 'noAuth',
+export interface UpdatePasswordFormType {
+  oldPassword: string;
+  newPassword: string;
+  confirmNewPassword: string;
+}
+
+/* Datasource types */
+export enum AuthType {
+  NoAuth = 'no_auth',
+  UsernamePasswordType = 'username_password',
 }
 
-export interface CreateNewCredentialType {
+export const credentialSourceOptions = [
+  { value: AuthType.UsernamePasswordType, inputDisplay: 'Username & Password' },
+  { value: AuthType.NoAuth, inputDisplay: 'No authentication' },
+];
+
+export interface DataSourceAttributes extends SavedObjectAttributes {
```

Comment:
wonder if you could directly use the type defined in data source plugin?

Reply:
I was doing the same. But it is always better to define types separately in each module so that they stay independent of each other.

I just had to make password as optional field. I cannot go to backend code and change that. So for simplicity and to avoid complications .. I copied the types and made necessary changes

Reply:
why only password can be optional but not user name?

Reply:
> I was doing the same. But it is always better to define types separately in each module so that they stay independent of each other.
> 
> I just had to make password as optional field. I cannot go to backend code and change that. So for simplicity and to avoid complications .. I copied the types and made necessary changes

Since DS plugin is the dependency of DSM, do we still need to copy the code?